File: app_software_ocrfactura/lib/models/factura.dart
class Factura {
  final int idFactura;
  final int idUsuario;
  final int idEmpresa;
  final String tipoComprobante;
  final String numeroComprobante;
  final String fechaEmision;
  final double subtotal;
  final double igv;
  final double total;

  Factura({
    required this.idFactura,
    required this.idUsuario,
    required this.idEmpresa,
    required this.tipoComprobante,
    required this.numeroComprobante,
    required this.fechaEmision,
    required this.subtotal,
    required this.igv,
    required this.total,
  });

  factory Factura.fromJson(Map<String, dynamic> json) {
    double toDouble(dynamic v) {
      if (v is num) return v.toDouble();
      return double.tryParse(v?.toString() ?? '0') ?? 0.0;
    }

    int toInt(dynamic v) {
      if (v is int) return v;
      return int.tryParse(v?.toString() ?? '0') ?? 0;
    }

    return Factura(
      idFactura: toInt(json['id_factura']),
      idUsuario: toInt(json['id_usuario']),
      idEmpresa: toInt(json['id_empresa']),
      tipoComprobante: (json['tipo_comprobante'] ?? '').toString(),
      numeroComprobante: (json['numero_comprobante'] ?? '').toString(),
      fechaEmision: (json['fecha_emision'] ?? '').toString(),
      subtotal: toDouble(json['subtotal']),
      igv: toDouble(json['igv']),
      total: toDouble(json['total']),
    );
  }
}

File: api_software_py/app/models/factura.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Date,
    DECIMAL,
    TIMESTAMP,
    ForeignKey,
    func
)

from sqlalchemy.orm import relationship

from app.database import Base


class Factura(Base):

    __tablename__ = "factura"


    id_factura = Column(
        Integer,
        primary_key=True,
        index=True
    )


    id_usuario = Column(
        Integer,
        ForeignKey("usuario.id_usuario"),
        nullable=False
    )


    id_empresa = Column(
        Integer,
        ForeignKey("empresa.id_empresa")
    )


    tipo_comprobante = Column(
        String(50)
    )


    numero_comprobante = Column(
        String(50)
    )


    fecha_emision = Column(
        Date
    )


    subtotal = Column(
        DECIMAL(10,2)
    )


    igv = Column(
        DECIMAL(10,2)
    )


    total = Column(
        DECIMAL(10,2)
    )


    imagen_url = Column(
        String(255)
    )


    estado = Column(
        Integer,
        nullable=False,
        server_default="1"
    )


    observaciones = Column(
        String(255)
    )


    fecha_registro = Column(
        TIMESTAMP,
        server_default=func.now()
    )


    # relación con detalle_factura
    detalles = relationship(
        "DetalleFactura",
        back_populates="factura",
        cascade="all, delete-orphan"
    )


    empresa = relationship("Empresa")